core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1967,6 +2004,23 @@ class SnappySession(_sc: SparkContext) extends SparkSession(_sc) {
     }
     (scalaTypeVal, SnappySession.getDataType(storeType, storePrecision, storeScale))
   }
+
+
+  SnappyContext.getClusterMode(_sc) match {
+    case ThinClientConnectorMode(_, _) => {
+      _sc.listenerBus.addListener(new AppSparkListener)
+    }
+    case _ => // Nothing to do
+  }
+}
```

Comment:
This will end up adding the same listener for every session and will end up creating too many listeners. See comment below.